fineract-progressive-loan/src/main/java/org/apache/fineract/portfolio/loanaccount/domain/transactionprocessor/impl/ChargeOrTransaction.java

```diff
@@ -58,7 +58,7 @@ private LocalDate getEffectiveDate() {
     }
 
     private boolean isBackdatedCharge() {
-        return loanCharge.get().getDueDate().isBefore(loanCharge.get().getSubmittedOnDate());
+        return (loanCharge.get().getDueDate() != null && loanCharge.get().getDueDate().isBefore(loanCharge.get().getSubmittedOnDate()));
```

Comment:
This change is unrelated. If this is a fix for something, please raise it in a new PR!

Reply:
Done! removed, It belongs to an issue when you have a Loan Installment Fee